src/test/java/org/wso2/carbon/connector/integration/test/StackExchangeConnectorIntegrationTest.java

```diff
@@ -0,0 +1,699 @@
+                    }
+                }
+                stackExchangeProperties.setProperty(
+                        STACKEXCHANGE_HAS_QUESTION_WITH_ANSWERS, String.valueOf(hasAcceptedAnswer));
+                if (!hasAcceptedAnswer) {
+                    Integer unacceptedAnswerId = placeHolderQuestionAnswerItems.getRandom(
+                            SE_RES_KEY_ANSWER_ID, Integer.class);
+                    connectorProperties.setProperty(PROP_KEY_UNACCEPTED_ANSWER_ID, String.valueOf(unacceptedAnswerId));
+                }
+            }
+            /* API routes like 'voting up a question' does not allow
+               user to up vote his/her own post. In those cases we should
+               avoid using placeHolderQId as the questionId. Hence looping
+               to guarantee it never happens. Length check is necessary to
+               avoid infinite looping. */
+            if (questionItems.length() > 1) {
+                int qid = Integer.parseInt(placeHolderQId);
+                while (questionId.equals(qid)) {
+                    questionId = questionItems.getRandom(SE_RES_KEY_QUESTION_ID, Integer.class);
+                }
+            }
+        }
+        connectorProperties.setProperty(PROP_KEY_QUESTION_ID, String.valueOf(questionId));
+
+        /* Save a random id of an answer */
+        StackExchangeItems answerItems = getStackExchangeAnswerItems(questionId);
+        Integer answerId = answerItems.getRandom(SE_RES_KEY_ANSWER_ID, Integer.class);
+        Integer[] answerIdArray = answerItems.getAll(SE_RES_KEY_ANSWER_ID, Integer.class);
+        connectorProperties.setProperty(PROP_KEY_ANSWER_ID, String.valueOf(answerId));
+        setListLikePropertyInConnector(PROP_KEY_ANSWER_ID_LIST, answerIdArray);
+
+        /* Save availability of Placeholder answer id */
+        if (StringUtils.isEmpty(placeHolderAId)) {
+            stackExchangeProperties.setProperty(STACKEXCHANGE_HAS_ANSWER, String.valueOf(false));
+        } else {
+            stackExchangeProperties.setProperty(STACKEXCHANGE_HAS_ANSWER, String.valueOf(true));
+            /* TODO: Check placeHolderAId and answer id collide. */
+            if (hasAnswerAccepted(Integer.parseInt(placeHolderAId), placeHolderASite)) {
+                stackExchangeProperties.setProperty(STACKEXCHANGE_IS_UNACCEPTED_ANSWER, String.valueOf(false));
+            } else {
+                stackExchangeProperties.setProperty(STACKEXCHANGE_IS_UNACCEPTED_ANSWER, String.valueOf(true));
+            }
+        }
+
+        /* Check credential availability to avoid unnecessary failures */
+        if (StringUtils.isNotEmpty(accessToken) && StringUtils.isNotEmpty(key)) {
+            /* Save privilege wordings belong to given credentials */
+            StackExchangeItems privilegeItems = getStackExchangePrivilegeItems(accessToken);
+            String[] privilegeShortDescriptionArray = privilegeItems.getAll(SE_RES_KEY_SHORT_DESCRIPTION, String.class);
```

Comment:
Formatting issue: line (120) is exceeded.

Reply:
Fix with 503a067